### PR TITLE
fix(core/biblio):  auto normalize references

### LINKED ISSUE #1535: auto normalize references
today, if you have:

```
[[html]] and <a data-cite="HTML">
```

Or variance in the above, ReSpec warns:

> "[HTML] is referenced in 2 ways: ('HTML', 'html'). This causes duplicate entries in the References section."

This is annoying for a number of reasons: 
 1. finding the culprit duplicate can be really hard in a large document. Specs can be thousands of lines long, and contain 50-100 citations. 
 1. Clearly, ReSpec knows the two are clashing - so it should just automatically fix the problem (just use the first encountered convention) and avoid adding duplicate entries to the bibliography.

### PR DESCRIPTION
**Pull request by sidvishnoi**
Fixes #1535

### CODE REVIEW

File: src/core/biblio.js [hunk 1]
```diff
@@ -220,8 +243,9 @@ async function updateFromNetwork(refs, options = { forceUpdate: false }) {
     return;
   }
   let response;
+  const refsToGet = [...new Set(refs.map(ref => ref.toLowerCase()))];
```
Comment (marcoscaceres) [inline src/core/biblio.js]: nit: s/refsToGet/refsToFetch

File: src/core/inlines.js [hunk 2]
```diff
@@ -61,44 +59,48 @@ export function run(conf) {
...
         ) {
           matched = matched.split(/\s+/).join(" ");
-          df.appendChild(hyperHTML`<em class="rfc2119" title="${matched}">${matched}</em>`);
+          df.appendChild(
+            hyperHTML`<em class="rfc2119" title="${matched}">${matched}</em>`
+          );
           // remember which ones were used
           conf.respecRFC2119[matched] = true;
-        } else if (/^\[\[/.test(matched)) {
+        } else if (matched.startsWith("[[")) {
```
Comment (marcoscaceres) [inline src/core/inlines.js]: 👍, same with the rest.

File: src/core/biblio.js [hunk 3]
```diff
@@ -129,50 +127,74 @@ function bibref(conf) {
...
+            ? conf.l10n.norm_references
+            : conf.l10n.info_references
+        }</h3>
       </section>`;
     addId(sec);
 
     refs.sort((a, b) =>
       a.toLocaleLowerCase().localeCompare(b.toLocaleLowerCase())
     );
+    refs = refs.map(getRefContent);
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/biblio.js]: rounds valid as with this assignment, the type is changing (from string to objects)
any variable name suggestion? (this is the hard stuff)

File: src/core/biblio.js [hunk 4]
```diff
@@ -129,50 +127,74 @@ function bibref(conf) {
...
+
+    const aliases = refs.reduce((aliases, { key, ref }) => {
+      aliases[key] = aliases[key] || [];
+      if (!aliases[key].includes(ref.toLowerCase())) aliases[key].push(ref);
+      return aliases;
+    }, {});
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/biblio.js]: The addition to `aliases{}` is conditional here. /me fails to see how can we use a map.
Am I missing something?

File: src/core/biblio.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/biblio.js]: refObjects?

File: src/core/biblio.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/biblio.js]: Maybe: 
```JS
const aliases = refs.reduce((aliases, { key, ref }) => {
  const keys = !aliases.has(key) ? aliases.set(key, []).get(key) : aliases.get(key);
  if (!keys.includes(ref.toLowerCase())) keys.push(ref);
  return aliases;
}, new Map());
```

[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/biblio.js]: I got confused in `map` vs `Map` lol

**Pull request merged**